systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicST.java

```diff
@@ -165,6 +167,88 @@ void testCreateTopicViaAdminClient() throws ExecutionException, InterruptedExcep
+                        .editTopicOperator()
+                            .withReconciliationIntervalSeconds(120)
+                        .endTopicOperator()
+                    .endEntityOperator()
+                .endSpec()
+                .done();
+
+        Properties properties = new Properties();
+
+        properties.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, KafkaResource.kafkaClient().inNamespace(NAMESPACE)
+                .withName(clusterName).get().getStatus().getListeners().stream()
+                .filter(listener -> listener.getType().equals(Constants.EXTERNAL_LISTENER_DEFAULT_NAME))
+                .findFirst()
+                .orElseThrow(RuntimeException::new)
+                .getBootstrapServers());
+
+        try (AdminClient adminClient = AdminClient.create(properties)) {
+
+            String topicName = "topic-create-delete-create";
+
+            KafkaTopicResource.topic(clusterName, topicName)
+                    .editSpec()
+                    .withReplicas(3)
+                    .endSpec()
+                    .done();
+            KafkaTopicUtils.waitForKafkaTopicReady(topicName);
+
+            adminClient.describeTopics(singletonList(topicName)).values().get(topicName);
+
+            for (int i = 0; i < 10; i++) {
+                Thread.sleep(2_000);
+                LOGGER.info("Iteration {}: Deleting {}", i, topicName);
+                cmdKubeClient().deleteByName(KafkaTopic.RESOURCE_KIND, topicName);
+                KafkaTopicUtils.waitForKafkaTopicDeletion(topicName);
+                TestUtils.waitFor("Deletion of topic " + topicName, 1000, 15_000, () -> {
+                    try {
+                        return !adminClient.listTopics().names().get().contains(topicName);
+                    } catch (ExecutionException | InterruptedException e) {
+                        return false;
+                    }
+                });
+                Thread.sleep(2_000);
+                long t0 = System.currentTimeMillis();
+                LOGGER.info("Iteration {}: Recreating {}", i, topicName);
+                KafkaTopicResource.topic(clusterName, topicName)
+                        .editSpec()
+                        .withReplicas(3)
+                        .endSpec()
+                        .done();
+                ResourceManager.waitForResourceStatus(KafkaTopicResource.kafkaTopicClient(), "KafkaTopic", NAMESPACE, topicName, Ready, 15_000);
+                TestUtils.waitFor("Recreation of topic " + topicName, 1000, 2_000, () -> {
+                    try {
+                        return adminClient.listTopics().names().get().contains(topicName);
+                    } catch (ExecutionException | InterruptedException e) {
+                        return false;
+                    }
+                });
+                if (System.currentTimeMillis() - t0 > 10_000) {
+                    fail("Took too long to recreate");
+                }
```

Comment:
I don't think this is needed. In case the topic is not recreated during the timeout period specified in `waitFor` above, the test will fail anyway. So in case, you will specify the timeout for 10s, it should be almost the same and the code will be shorter. Unless there is something I don't see.

Reply:
It is needed. `System.currentTimeMillis() - t0 ` is measuring the time for:

* Kube to create the `KafkaTopic` (admittedly this should be negligable)
* The time for the TO to react to that, creating the topic and setting the `Ready` status
* The topic to actually exist in the broker's metadata cache.

The `waitFor` timeout is measuring only the last bullet.

Without this assertion the test always passes, even without the changes in this PR which fix the issue. With this assertion (but without the changes in this PR which fix the issue) it fails on the second iteration of the `for` loop because the TO wrongly retains some state about the topic having existed in the past, and so the TO has to wait for a full reconciliation before that wrong state gets sorted out (well, it's ignored I think).

Reply:
I see, make sense